spec/controllers/v1.0/requests_controller_spec.rb

```diff
@@ -40,49 +48,111 @@
         expect(json['data'].size).to eq(5)
       end
     end
+
+    context 'when approver' do
+      let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => false, :approver? => true, :owner? => false) }
+      before do
+        allow(RBAC::Access).to receive(:new).with('requests', 'read').and_return(access_obj)
+        allow(access_obj).to receive(:process).and_return(access_obj)
+      end
+
+      it 'returns status code 403' do
+        get "#{api_version}/workflows/#{workflow_id}/requests", :headers => request_header
+        expect(response).to have_http_status(403)
+      end
+    end
+
+    context 'when owner' do
+      let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => false, :approver? => false, :owner? => true) }
+      before do
+        allow(RBAC::Access).to receive(:new).with('requests', 'read').and_return(access_obj)
+        allow(access_obj).to receive(:process).and_return(access_obj)
+
+        get "#{api_version}/workflows/#{workflow_id}/requests", :headers => request_header
+      end
+
+      it 'returns status code 403' do
+        expect(response).to have_http_status(403)
+      end
+    end
   end
 
   # Test suite for GET /requests
   describe 'GET /requests' do
-    before { get "#{api_version}/requests", :params => { :limit => 5, :offset => 0 }, :headers => request_header }
+    before do
+      allow(RBAC::Access).to receive(:new).with('requests', 'read').and_return(access_obj)
+      allow(access_obj).to receive(:process).and_return(access_obj)
+      allow(access_obj).to receive(:approver_id_list).and_return([])
+      allow(access_obj).to receive(:owner_id_list).and_return([])
 
-    it 'returns requests' do
-      expect(json['links']).not_to be_empty
-      expect(json['links']['first']).to match(/limit=5&offset=0/)
-      expect(json['links']['last']).to match(/limit=5&offset=5/)
-      expect(json['data'].size).to eq(5)
+      get "#{api_version}/requests", :params => { :limit => 5, :offset => 0 }, :headers => request_header
```

Comment:
in general all `get` commands should be placed inside `it` block rather in `before` block. You may want to create a separate PR to do the move all for occurrences.